custom/enikshay/reports/web_dashboard.py

```diff
@@ -0,0 +1,282 @@
+from corehq.apps.reports.graph_models import PieChart, MultiBarChart, Axis
+from corehq.apps.reports_core.filters import Choice
+from corehq.apps.style.decorators import use_nvd3
+from corehq.apps.userreports.models import StaticReportConfiguration
+from corehq.apps.userreports.reports.factory import ReportFactory
+from custom.enikshay.reports.filters import EnikshayLocationFilter
+from custom.enikshay.reports.generic import EnikshayReport
+from custom.enikshay.reports.sqldata.case_finding_sql_data import CaseFindingSqlData
+from custom.enikshay.reports.sqldata.charts_sql_data import ChartsSqlData
+from custom.enikshay.reports.sqldata.treatment_outcome_sql_data import TreatmentOutcomeSqlData
+from dimagi.utils.decorators.memoized import memoized
+
+from django.utils.translation import ugettext_lazy, ugettext as _
+
+
+class WebDashboardReport(EnikshayReport):
+
+    name = ugettext_lazy('Web Dashboard')
+    report_title = ugettext_lazy('eNikshay Report Dashboard')
+    slug = 'web_dashboard'
+    use_datatables = False
+    report_template_path = 'enikshay/web_dashboard.html'
+    fields = (DatespanFilter, EnikshayLocationFilter)
+
+    @use_nvd3
+    def decorator_dispatcher(self, request, *args, **kwargs):
+        return super(WebDashboardReport, self).decorator_dispatcher(request, *args, **kwargs)
+
+    @property
+    def headers(self):
+        return DataTablesHeader()
+
+    @property
+    def rows(self):
+        return []
+
+    @property
+    def report_context(self):
+        report_context = super(WebDashboardReport, self).report_context
+        report_context['total_patients'] = self.charts_sql_data[0].get('total_patients', {}).get('sort_key', 0)
+        return report_context
+
+    @property
+    @memoized
+    def case_finding_sql_data(self):
+        return CaseFindingSqlData(config=self.report_config).get_data()
+
+    @property
+    @memoized
+    def charts_sql_data(self):
+        return ChartsSqlData(config=self.report_config).get_data()
+
+    @property
+    @memoized
+    def treatment_outcome_sql_data(self):
+        return TreatmentOutcomeSqlData(config=self.report_config).get_data()
+
+    @property
+    def charts(self):
+        case_finding_sql_data = self.case_finding_sql_data[0]
+        sputum_conversion_report = ReportFactory.from_spec(
+            StaticReportConfiguration.by_id('static-%s-sputum_conversion' % self.domain), include_prefilters=True
+        )
+
+        filter_values = {}
+
+        if self.datespan:
+            filter_values['date'] = self.datespan
+
+        locations_id = [
+            Choice(value=location_id, display='') for location_id in self.report_config.locations_id
+            if location_id
+        ]
+
+        if locations_id:
+            filter_values['village'] = locations_id
+
+        sputum_conversion_report.set_filter_values(filter_values)
+        sputum_conversion_data = sputum_conversion_report.get_data()[0]
```

Comment:
nit: everything from `sputum_conversion_report = ReportFactory.from_spec(...` to here might be a good candidate to get pulled out into a helper function